### ISSUE TITLE
Randomize Training data

### ISSUE THREAD

**Issue by calicratis19** [xiph/rnnoise#43]
My training data set is a 50000000 X 87 matrix. Each iteration my model reads the data in the same way. Its recommended to shuffle training data at the beginning of each epoch so that it generalizes better. Is it possible to shuffle the training data? Will it somehow make the data invalid?

**Comment by jmvalin** [comment]
See https://keras.io/models/model/
By default, Keras will shuffle the data for you when training.

**Comment by calicratis19** [comment]
If I shuffle the rows randomly will it be alright? Is there any restriction on the shuffling?

**Comment by wuqiangch** [comment]
@calicratis19 @jmvalin ,I want to train the model using my data .Where can I find the code to generate the input data?Thanks!

**Comment by calicratis19** [comment]
@wuqiangch 

Steps:

(1) cd src ; ./compile.sh

(2) ./denoise_training signal.raw noise.raw count > training.f32

    (note the matrix size and replace 50000000 87 below)

(3) cd training ; ./bin2hdf5.py ../src/training.f32 50000000 87 training.h5

(4) ./rnn_train.py

(5) ./dump_rnn.py weights.hdf5 rnn_data.c rnn_data.rnnn name

**Comment by carclodefly** [comment]
@calicratis19  Dear sir. what shoud I do to prepare the  signal.raw and noise.raw,. how much word did you do on the datas that‘s jmvalin operatored to make it match this step， thank you ！

**Comment by carclodefly** [comment]
@calicratis19  I mean ,how can i prepare the signal.raw and noise.raw from pcm or wav datas.

**Comment by calicratis19** [comment]
@carclodefly you need merge all your noise audio files to a single file and all noiseless audi files to another single file. Then convert them to pcm format and name them noise.raw and signal.raw.

**Comment by carclodefly** [comment]
> 
> 
> @carclodefly you need to merge all your noise audio files to a single file and all noiseless audio files to another single file. Then convert them to pcm format and name them noise.raw and signal.raw.

thank you very much ！

**Comment by carclodefly** [comment]
@calicratis19  and，Sir. if I wanna train my model with the noise datas that Mr. jmvalin provided， what kind of signal none noise data I can use ？and how much signal data I need ？

**Comment by carclodefly** [comment]
> 
> 
> @carclodefly you need to merge all your noise audio files to a single file and all noiseless audio files to another single file. Then convert them to pcm format and name them noise.raw and signal.raw.

Sir. I do like this with the signal form McGill TSP speech database，and the noise  Mr. jmvalin offered， but the model output is not good. Do you know how to deal with the signal ？I found the feature extraction only simply add signal and noise on Frequency domain point.

**Comment by calicratis19** [comment]
@carclodefly I also could not make the model work better. My output was worse than the original one. I could not get the model decrease the loss much. So I gave up.

**Comment by sporwar-lifesize** [comment]
> @wuqiangch
> 
> Steps:
> 
> (1) cd src ; ./compile.sh
> 
> (2) ./denoise_training signal.raw noise.raw count > training.f32
> 
> ```
> (note the matrix size and replace 50000000 87 below)
> ```
> (3) cd training ; ./bin2hdf5.py ../src/training.f32 50000000 87 training.h5
> 
> (4) ./rnn_train.py
> 
> (5) ./dump_rnn.py weights.hdf5 rnn_data.c rnn_data.rnnn name

Step 2, you mention the third argument as count. Is this the <ouput denoised> as you see it in the usage.  or shouldn't this be speech.raw noise.raw output.f32 as mentioned in #8 
usage: ./denoise_training <speech> <noise> <output denoised>

**Comment by sporwar-lifesize** [comment]
I am facing some issues when I tried to play around with my custom speech and noise data. I have a min of both speech and noise (16-bit, 48Khz, mono) PCM files that I am passing to denoise_training.  I see 50000000 x 87  as the matrix size. But when I run the third command, I get File "./bin2hdf5.py", line 9, in <module>   MemoryErrror.  Appreciate insights on to move forward. Thanks

**Comment by momo1986** [comment]
> name

@sporwar-lifesize , For command "./dump_rnn.py weights.hdf5 rnn_data.c rnn_data.rnnn name", where it the stored path for "name"?

I think "python dump_rnn.py" makes more sense here, isn't it?
